packages/react/src/components/TagButton/index.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { HTMLProps } from 'react';
+import Icon, { IconType } from '../Icon';
+import { ContentNode } from '../../types';
+import Tag, { TagLabel } from '../Tag';
+import classNames from 'classnames';
+import Button from '../Button';
+
+interface TagButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
+  label: ContentNode;
+  value: ContentNode;
+  icon: IconType;
+  action: (e: React.MouseEvent<HTMLButtonElement>) => void;
```

Comment:
What's the expected behavior if someone sets props on the component like this?

```jsx
<TagButton action={...} onClick={...} />
```

It looks like we allow both via the type, but in the implementation below we clobber `onClick` with `action`. If the desired behavior is the action is only set via that prop, we likely need to omit `onClick`:

```tsx
interface TagButtonProps extends Omit<React.HTMLAttributes<HTMLButtonElement>, 'onClick'>
```